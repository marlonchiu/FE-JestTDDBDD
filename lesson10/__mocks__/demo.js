export const fetchData = () => {
  return new Promise((resolve, reject) => {
    resolve("(function(){return '123'})()")
  })
}

// 假设返回数据是
// {
//   data: '(function(){return '123'})()'
// }
