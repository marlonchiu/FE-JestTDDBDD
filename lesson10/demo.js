import axios from 'axios'

export const fetchData = () => {
  return axios.get('/').then(res => res.data)
}

// 假设返回数据是
// {
//   data: '(function(){return '123'})()'
// }

export const getNumber = () => {
  return 123
}
