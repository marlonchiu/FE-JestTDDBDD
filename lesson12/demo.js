import Util from './util'

const demoFunction = (a, b) => {
  const util = new Util()
  // 模拟调用了 a b function
  util.a(a)
  util.b(b)
}

export default demoFunction
