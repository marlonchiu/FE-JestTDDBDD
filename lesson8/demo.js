import axios from 'axios'

export const runCallback = (callback) => {
  callback('hello')
}

export const createObject = (classItem) => {
  new classItem()
}

export const getData = () => {
  return axios.get('/api').then(res => res.data)
}
