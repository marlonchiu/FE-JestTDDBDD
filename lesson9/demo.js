export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: 8080,
    domain: 'localhost',
    time: '2019',
    time2: new Date()
  }
}

export const generateAnotherConfig = () => {
  return {
    server: 'http://localhost',
    port: 8080,
    domain: 'localhost',
    time: '2020',
    time2: new Date()
  }
}
