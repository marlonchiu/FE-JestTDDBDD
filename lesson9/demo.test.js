import { generateConfig, generateAnotherConfig } from "./demo";

/**
 * 快照
 * toMatchSnapshot 生成快照文件夹__snapshots__
 * toMatchInlineSnapshot  行内快照
 *
 * 对于变化的配置量 Snapshot({ time2: expect.any(Date)})
 *   同样可以接收 Date String Number
 *
 */
// test('should generateConfig 函数', () => {
//   // expect(generateConfig()).toEqual({
//   //   server: 'http://localhost',
//   //   port: 8080,
//   //   domain: 'localhost'
//   // })
//   expect(generateConfig()).toMatchSnapshot({
//     time2: expect.any(Date)
//   })
// })

// test('should generateAnotherConfig 函数', () => {
//   expect(generateAnotherConfig()).toMatchSnapshot({
//     time2: expect.any(Date)
//   })
// })

test("行内 generateAnotherConfig 函数", () => {
  expect(generateAnotherConfig()).toMatchInlineSnapshot(
    {
      time2: expect.any(Date),
    },
    `
    Object {
      "domain": "localhost",
      "port": 8080,
      "server": "http://localhost",
      "time": "2020",
      "time2": Any<Date>,
    }
  `
  );
});
