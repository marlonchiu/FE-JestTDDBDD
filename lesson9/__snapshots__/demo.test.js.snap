// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generateAnotherConfig 函数 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": "2020",
  "time2": Any<Date>,
}
`;

exports[`should generateConfig 函数 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": "2019",
  "time2": Any<Date>,
}
`;
